source/docs/prismic/routable-types.md

```diff
@@ -0,0 +1,220 @@
+
+If no GraphQL type was created to represent the custom data type within Front-Commerce, please create a GraphQL type to represent the custom type in schema.gql of the server module's directory
+
+```diff
++type Album implements Routable {
++  url: String # the newly added field
++  artist: String
++  title: String
++  release_date: Date
++  path: String # for the routable interface
++}
+```
+
+Note that in order for the custom type to be routable it needs to implement the core Front-Commerce [Routable interface](https://gitlab.com/front-commerce/front-commerce/-/blob/2.7.1/src/server/modules/front-commerce/core/schema.gql#L87). That is why the GraphQL type must have a field called `path` to represent the URL of the routable type.
+
+Add a new GraphQL type to the module's resolver:
+
+```diff
+export default {
+  Query: {
+    ...
+  },
++  Album: {
++    path: async (content, _, { loaders }) => {
++      const baseUrl = (await loaders.Shop.getCurrentShop()).baseUrl,
++      return `{baseUrl}/${content.url}`;
++    },
++  },
++};
+```
+
+Note we added a path field resolver that appends the shop's baseUrl  to the `url` field of our custom Prismic type.
+
+## Register the Prismic custom type as a routable type
+
+To make a custom Prismic type routable it must be registered with the Prismic module. First ensure that the `Prismic/core` is added as a module's dependency then use `loaders.Prismic.registerRoutableType` to add the custom type as routable type:
+
+```diff
+import typeDefs from "./schema.gql";
+import resolvers from "./resolvers";
+
+export default {
+  namespace: "<module_name_space>",
+-  dependencies: [],
++  dependencies: ["Prismic/Core"],
+  typeDefs,
+  resolvers,
+  contextEnhancer: ({ req, loaders }) => {
+
++    const { TitleTransformer, DateTransformer } = loaders.Prismic.transformers;
++    const contentTransformOptions = {
++      fieldTransformers: {
++        title: new TitleTransformer(),
++        release_date: new DateTransformer(),
++      },
++    };
++
++    loaders.Prismic.registerRoutableType({
++      typeIdentifier: "album", // <-- Prismic custom type
++      urlFieldName: "url",     // <-- Prismic API ID mentioned above
++      graphQLType: "Album",    // <-- GraphQL type created above
++      contentTransformOptions,
++      isSitemapable: false,
++      postTransformer: (url, document) => { // optional function postTransformer
++        if(document.isPublished) {  // possible usecase
++          return document;
++        }
++      }
++    });
+
+    return {};
+  },
+};
+```
+
+Note the `postTransformer` above is optional. It is a function that will be called after the transformation is done using `contentTransformOptions`. it is given the current URL being resolved and the transformed document. It can be used if you have some custom logic to apply to the transformed document or if you want to supress a document from showing using some custom logic (returning a falsey value).
```

Comment:
```suggestion
Note the `postTransformer` above is optional. It is a function that will be called after the transformation is done using `contentTransformOptions`. It is given the current URL being resolved and the transformed document. It can be used if you have some custom logic to apply to the document or if you want to suppress a document from showing using some custom logic (returning a _falsy_ value).
```